### PR TITLE
feat: Chip

### LINKED ISSUE #6: Chips
[MD Docs Link](https://material.google.com/components/chips.html#chips-specs)

### PR DESCRIPTION
**Pull request by gawrysiak**
Adds simple chip component. Implements [Material Design Guidelines for Chips](https://material.io/guidelines/components/chips.html). Features:
- standard text chips
- icon / image chips
- delete function / icon
- optional onPress

screen shot 2018-05-05 at 21 57 41

Free avatar icon taken from iconfinder.

### Motivation

Fixes #6. If this one's okay, it could be nice to implement some sort of a `ChipInput` component as well.

### Test plan

Snapshot tests added. Run `yarn test`.

### CODE REVIEW

**Comment by satya164** [comment]
Thanks!!!

Can you change `text` to `children`? I'll take a proper look tomorrow.

**Comment by gawrysiak** [comment]
@satya164 should have seen that coming, it's changed now :) :+1:

File: src/components/Chip.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+import { View, StyleSheet, TouchableWithoutFeedback } from 'react-native';
+import color from 'color';
+import Icon from './Icon';
+import Text from './Typography/Text';
+import { black, white } from '../styles/colors';
+import withTheme from '../core/withTheme';
+import type { Theme } from '../types';
+import type { IconSource } from './Icon';
+
+type Props = {
```
[the remark this answers is not in the document]
Comment (gawrysiak) [reply src/components/Chip.js]: ready :+1:

File: src/components/Chip.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  children: string | Array<string>,
+  icon?: IconSource,
+  onPress?: Function,
```
Comment (Trancever) [inline src/components/Chip.js]: Could you change "Function" to "() => mixed". We follow this pattern in other componets.

File: src/components/Chip.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  onDelete?: Function,
```
Comment (Trancever) [inline src/components/Chip.js]: "Function" to "() => mixed".

File: src/components/Chip.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+   */
+  children: string | Array<string>,
+  /**
+   * Name of the icon. Can be a string, an image source or a react component.
+   */
+  icon?: IconSource,
+  /**
+   * Function to execute on press.
+   */
+  onPress?: Function,
```
Comment (Trancever) [inline src/components/Chip.js]: Could you change "Function" to "() => mixed". We follow this pattern in other components.

Comment (Trancever) [inline src/components/Chip.js]: Could you change "Function" to "() => mixed". We follow this pattern in other componets.

File: src/components/Chip.js [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+  /**
+   * Function to execute on delete.
+   */
+  onDelete?: Function,
```
Comment (Trancever) [inline src/components/Chip.js]: "Function" to "() => mixed"

Comment (gawrysiak) [reply src/components/Chip.js]: are you sure about this one? there's 19 occurrences for `Function` and none for `() => mixed`

Comment (satya164) [reply src/components/Chip.js]: We should change those `Function` to `() => mixed` xD

`() => mixed` makes it clear that the function receives so arguments while `Function` doesn't.

Comment (gawrysiak) [reply src/components/Chip.js]: ready :+1:

**Comment by satya164** [comment]
@gawrysiak I'll handle it.

**Comment by satya164** [comment]
Merged in b9aa79503c1aa8779ebd08190fc71d64fe93d0bf

**Pull request closed without merging**